SteakBot.Core/EventHandlers/CustomMessageHandlers/CommandMessageHandlers/CustomCommandMessageHandler.cs

```diff
@@ -1,32 +1,28 @@
-﻿using System.Collections.Generic;
-using System.Configuration;
-using System.IO;
-using System.Linq;
+﻿using System.Linq;
 using Discord;
 using Discord.WebSocket;
-using Newtonsoft.Json;
-using SteakBot.Core.Objects;
+using SteakBot.Core.Services;
 using SteakBot.Core.Objects.Enums;
 
 namespace SteakBot.Core.EventHandlers.CustomMessageHandlers.CommandMessageHandlers
 {
     internal class CustomCommandMessageHandler : BaseCommandMessageHandler
     {
-        private static readonly string MemeCommandsFileName = ConfigurationManager.AppSettings["memeCommandsRelativeFilePath"];
+        private readonly MemeService _memeService;
 
-        internal static IList<MemeCommand> Commands { get; } = LoadCommands();
-
-        public CustomCommandMessageHandler()
+        public CustomCommandMessageHandler(MemeService memeService)
         {
-            CommandNames = Commands.Select(x => x.Name);
+            _memeService = memeService;
+            _memeService.ReloadCommandsEvent += ReloadCommandNames;
+            ReloadCommandNames();
         }
 
         protected override bool InvokeInner(SocketUserMessage message)
         {
             var channel = message.Channel;
             var commandText = message.Content.Replace(CommandChar, "").Replace(DeleteMessageChar, "");
-            var command = Commands.Single(x => x.Name == commandText);
-            switch (command.ResultType)
+            var command = _memeService.MemeCommands.SingleOrDefault(x => x.Name == commandText);
```

Comment:
The `InvokeInner` is only called once we've verified that the command can be handled by this handler, so there shouldn't be any scenario where this lookup fails.
Can safely retyrn this to `.Single()` and remove the null propagation on the next line.